pkg/componentreadiness/jiraintegrator/jiraintegrator.go

```diff
@@ -0,0 +1,524 @@
+	views        []crtype.View
+	releases     []query.Release
+	sippyURL     string
+}
+
+func NewJiraIntegrator(jiraClient jirautil.Client, bqClient *bqclient.Client, prowURL, gcsBucket string,
+	cacheOptions cache.RequestOptions, views []crtype.View, releases []query.Release, sippyURL string) (JiraIntegrator, error) {
+	j := JiraIntegrator{
+		jiraClient:   jiraClient,
+		bqClient:     bqClient,
+		prowURL:      prowURL,
+		gcsBucket:    gcsBucket,
+		cacheOptions: cacheOptions,
+		releases:     releases,
+		sippyURL:     sippyURL,
+	}
+	if bqClient == nil || bqClient.BQ == nil {
+		return j, fmt.Errorf("we don't have a bigquery client for jira integrator")
+	}
+
+	if bqClient.Cache == nil {
+		return j, fmt.Errorf("we don't have a cache configured for jira integrator")
+	}
+
+	for _, v := range views {
+		if v.JiraIntegration.Enabled {
+			j.views = append(j.views, v)
+		}
+	}
+	return j, nil
+}
+
+func (j JiraIntegrator) getRequestOptionForView(view crtype.View) (crtype.RequestOptions, error) {
+	baseRelease, err := componentreadiness.GetViewReleaseOptions("basis", view.BaseRelease, j.cacheOptions.CRTimeRoundingFactor)
+	if err != nil {
+		return crtype.RequestOptions{}, err
+	}
+
+	sampleRelease, err := componentreadiness.GetViewReleaseOptions("sample", view.SampleRelease, j.cacheOptions.CRTimeRoundingFactor)
+	if err != nil {
+		return crtype.RequestOptions{}, err
+	}
+
+	variantOption := view.VariantOptions
+	advancedOption := view.AdvancedOptions
+
+	// Get component readiness report
+	reportOpts := crtype.RequestOptions{
+		BaseRelease:    baseRelease,
+		SampleRelease:  sampleRelease,
+		TestIDOption:   view.TestIDOption,
+		VariantOption:  variantOption,
+		AdvancedOption: advancedOption,
+		CacheOption:    j.cacheOptions,
+	}
+	return reportOpts, nil
+}
+
+func (j JiraIntegrator) getComponentReportForView(view crtype.View) (crtype.ComponentReport, error) {
+	reportOpts, err := j.getRequestOptionForView(view)
+	if err != nil {
+		return crtype.ComponentReport{}, fmt.Errorf("failed to get request option for view %s with error %v", view.Name, err)
+	}
+
+	report, errs := componentreadiness.GetComponentReportFromBigQuery(j.bqClient, j.prowURL, j.gcsBucket, reportOpts)
+	if len(errs) > 0 {
+		var strErrors []string
+		for _, err := range errs {
+			strErrors = append(strErrors, err.Error())
+		}
+		return crtype.ComponentReport{}, fmt.Errorf("component report generation encountered errors: " + strings.Join(strErrors, "; "))
+	}
+	return report, nil
+}
+
+func getProbabilityString(status crtype.Status, fisherExact float64) string {
+	if status == crtype.SignificantRegression || status == crtype.ExtremeRegression {
+		return fmt.Sprintf("Probability of significant regression: %.2f%%\n\n", (1-fisherExact)*100)
+	} else if status == crtype.SignificantImprovement {
+		return fmt.Sprintf("Probability of significant improvement: %.2f%%\n\n", (1-fisherExact)*100)
+	}
+	return "There is no significant evidence of regression\n\n"
```

Comment:
I realized in my pass rate PR where sometimes we won't be returning fisher results, perhaps the server should provide an "explanation" string in the regressed tests API response with this kind of text, so we don't have to fork the same logic in multiple places. I'm going to test adding that to the pass rates PR and if we do we can use that here so we just render a human readable explanation provided by the server. It's going to get complicated otherwise. I'm going to try that out and maybe we can leverage here.

Reply:
I might need to take care of this tomorrow.